src/tests/ftest/osa/dmg_negative_test.py

```diff
@@ -38,9 +38,8 @@ def validate_results(self, exp_result, dmg_output):
         """
         if exp_result == "Pass":
             # Check state before hand as wait for rebuild
-            # does not consider the idle state
             state = self.get_rebuild_state()
-            if state not in ("done", "idle"):
+            if state not in "done":
```

Comment:
When `in` is used with a string it will match on any character in the string, what you really want is this:
```suggestion
            if state is not "done":
```

Reply:
![image](https://user-images.githubusercontent.com/48604608/153456333-320ee6a1-268c-4cee-9147-8ce94efa5e6e.png)

Flake recommends to use "==".